src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,140 @@
+package duke;
+
+import duke.exceptions.DukeException;
+import duke.task.Task;
+import duke.ui.UiController;
+import javafx.application.Application;
+import javafx.stage.Stage;
+
+import java.util.List;
+
+/**
+ * The main driver class for Duke chatbot. This bot handles high level logic for the bot including startup and
+ * shutdown.
+ */
+public class Duke extends Application {
+    private final UiController ui;
+    private final TaskList taskList;
+    private final Storage storage;
+    private final Parser parser;
+
+
+    /**
+     * Primary constructor for an instance of the chatbot. This initialises the Ui, taskList, storage and parser
+     * instances used by the bot, which are not to be modified throughout the lifecycle of the bot.
+     */
+    public Duke() {
+        this.ui = new UiController(this);
+        this.taskList = new TaskList();
+        this.storage = new Storage("src/data/duke.txt", this.taskList);
+        this.parser = new Parser();
+    }
+
+    /**
+     * Terminates the current instance of the chatbot by exiting the program after displaying exit message.
+     */
+    void exit() {
+        storage.updateData(this.taskList);
+        ui.displayGoodbyeMessage();
+        System.exit(0);
+    }
+
+    private String taskAddedMessage(Task task) {
+        return "Got it. I've added this task:\n" +
+                task.toString() +
+                "\nNow you have " +
+                taskList.size() +
+                " tasks in the list\n";
+    }
+
+    /**
+     * Returns a <code>Message</code> that represents the response of the chatbot when prompted by an input by user.
```

Comment:
I think for the description try not to include syntax like the code syntax. I believe this part of the JavaDocs is just supposed to tell us in general what the function does, not what it returns. Otherwise, well done! I appreciate the fact that you put in a great amount of effort into your JavaDocs